lib/cdo/code_generation.rb

```diff
@@ -0,0 +1,29 @@
+module CodeGeneration
+  CONSONANTS = ('A'..'Z').to_a - %w(A E I O U)
+  BANNED_SUBSTRINGS = %w(CNT DCK DMN FCK PNS PSS SHT TTS)
+  MAX_ATTEMPTS = 100
+
+  # Generates a semi-random consonant string of the specified length,
+  # excluding banned substrings.
+  # @param length [Integer] length of returned random code (default 6)
+  # @param reject_if [Proc] return true if a code should be rejected and replaced by a new one (optional)
+  # @return [String] generated code
+  def self.random_code(length = 6, reject_if: nil)
```

Comment:
You can probably make a method to generate an unused code that takes either a block or a model name / param. More DRY